browser/src/client_win.cc

```diff
@@ -28,19 +28,54 @@ ClientWin::ClientWin(Config config) : ClientBase(std::move(config)) {
     return;
 
   pipename_ = pipename;
+  if (ConnectToPipe(pipename_, &hPipe_) != ERROR_SUCCESS) {
+    Shutdown();
+  }
 }
 
-DWORD ClientWin::ConnectToPipe(HANDLE* handle) {
+int ClientWin::Send(const ContentAnalysisRequest& request,
+                    ContentAnalysisResponse* response) {
+  // TODO: avoid extra copy.
+  ChromeToAgent chrome_to_agent;
+  *chrome_to_agent.mutable_request() = request;
+  bool success = WriteMessageToPipe(hPipe_,
+                                    chrome_to_agent.SerializeAsString());
+  if (success) {
+    std::vector<char> buffer = ReadNextMessageFromPipe(hPipe_);
+    AgentToChrome agent_to_chrome;
+    success = agent_to_chrome.ParseFromArray(buffer.data(), buffer.size());
+    if (success) {
+      *response = std::move(*agent_to_chrome.mutable_response());
+    }
+  }
+
+  return success ? 0 : -1;
+}
+
+int ClientWin::Acknowledge(const ContentAnalysisAcknowledgement& ack) {
+  // TODO: avoid extra copy.
```

Comment:
How about a more explicit TODO to switch the `const T&` override types to `T` so no extra copies are made?

Better yet, how about `T&&` since we're not bound by the regular Chrome style guide :) ?